### ISSUE TITLE
fortios_system_global missing vdom_mode attribute

### ISSUE THREAD

**Issue by poroping** [fortinetdev/terraform-provider-fortios#134]
vdom_admin moved to vdom_mode with the support of split-vdoms.

This resource is missing the vdom_mode attibute

```

fgt01 (global) # set v
--
vdom-mode                               Enable/disable support for split/multiple virtual domains (VDOMs).
vip-arp-range                           Controls the number of ARPs that the FortiGate sends for a Virtual IP (VIP) address range.
virtual-switch-vlan                     Enable/disable virtual switch VLAN.

fgt01 (global) # set vdom-mode
--
no-vdom       Disable split/multiple VDOMs mode.
split-vdom    Enable split VDOMs mode.
multi-vdom    Enable multiple VDOMs mode.

```

**Comment by frankshen01** [comment]
Hi @poroping , Thank you for your continued high-quality and insightful issues/requirements! Could you provide me your FortiGate firemware version? (GUI: Dashboard->Status->System Information->Fireware or CLI: get system status->Version) Thanks!

**Comment by poroping** [comment]
Hey, @frankshen01, no worries, thanks for the quick fixes. It's a 1800F running a non-merged build so maybe it's just me. Waiting on 6.4 to drop for 1800.

```Version: FortiGate-1800F v6.2.5,build5878,201021 (GA)```

**Comment by frankshen01** [comment]
Hi @poroping , Thank you for your information. I carefully checked the API official documentation of the FOS version you mentioned and the latest FOS versions. This parameter is not provided by the API official. I don't know if they will cancel or modify this parameter in the future. Can you please configure it in the following way?

```hcl
variable vdom_mode {
  type        = string
  default     = "split-vdom" #"no-vdom", "split-vdom", "multi-vdom"
}

resource "fortios_system_global" "trname" {
  admin_sport = 443
  alias       = "FGVM02TM20003062"
  hostname    = "ste11"
  timezone    = "04"
}

resource "fortios_json_generic_api" "test2" {
  path   = "/api/v2/cmdb/system/global"
  method = "PUT"
  json   = <<EOF
{
    "vdom-mode": "${var.vdom_mode}"
}
EOF
}

```
Thanks!

**Comment by poroping** [comment]
Hi @frankshen01 this command is documented in the 6.2.5 API documentation on FNDN, not sure if you have separate internal documentation but it's definitely on FNDN.

https://fndn.fortinet.net/index.php?/fortiapi/1-fortios/641/1/system/

I can run these tomorrow if needed, stuck in meetings rest of day.

**Comment by frankshen01** [comment]
@poroping , you are right, sorry I just realized that I searched with the wrong keyword. I will add this feature in the next version.

**Comment by poroping** [comment]
All good, there's no rush, this can wait for the merge with the beta in
1.9.0 I saw you have planned.

**Comment by frankshen01** [comment]
Feature supported and released, please see the latest version(v1.9.0): https://registry.terraform.io/providers/fortinetdev/fortios/latest and https://registry.terraform.io/providers/fortinetdev/fortios/latest/docs/resources/fortios_system_global `vdom_mode` for details, Thanks!

**Comment by poroping** [comment]
Awesome, thanks.
